src/main/webapp/app/shared/markdown-editor/commands/orderedListCommand.ts

```diff
@@ -8,54 +8,69 @@ export class OrderedListCommand extends Command {
 
     /**
      * @function execute
-     * @desc Use the markdown language for creating/removing an ordered list
+     * @desc Use the Markdown language for creating/removing an ordered list
      */
     execute(): void {
         const selectedText = this.getSelectedText();
         this.splitText(selectedText);
     }
 
     /**
+     * Splits the text and performs the necessary manipulations.
      * @function splitText
-     * @desc 1. Split the text at the line break into an array
-     *       2. Assign each line the position it has in the array
-     *       3. Call for each textLine the replaceText method
      * @param selectedText the selected text by the cursor
      */
     splitText(selectedText: string): void {
         const parseArray = selectedText.split('\n');
-        let addAmount = parseArray.length - 1;
-        for (const element of parseArray) {
-            this.replaceText(element, parseArray.length - addAmount);
-            addAmount--;
-        }
+        let manipulatedText = '';
+        let position = 1;
+
+        parseArray.forEach((line, index) => {
+            // Empty line -> we start again
+            if (line === '') {
```

Comment:
as noted https://github.com/ls1intum/Artemis/pull/5072#pullrequestreview-971134377 in markdown an unordered list doesn't reset until there is a line with some content (at least `&nbsp;`) that is not part of the list so resetting the list numbering here is misleading.

Reply:
I had a look at the CommonMark specification and it looks like there can be an arbitrary number of empty lines between list elements (https://spec.commonmark.org/0.30/#lists, Section 5.3, first paragraph: ‘The list items may be separated by any number of blank lines.’). Therefore, the current behaviour of not resetting the numbering in the preview is actually correct.

Adapting the current numbering implementation to keep empty lines as-is but not increase the index should not be a big change.

Reply:
I changed the logic to be similar to GitHub:


```
Test

Test
```
will result in (and vice versa):

```
1. Test
2.
3. Test
```

Is that what you would expect?

Reply:
![image](https://user-images.githubusercontent.com/60005702/168273490-05c49e0b-ab52-46a7-b069-fcd99ef471b3.png)

But i guess the Preview has still some issues which should be addressed in another PR

Reply:
Thanks, but I think that GitHub’s implementation in this case is not optimal. I think that it would be closest to the specification if
```
first item

second item
```
would be turned into
```
1. first item

2. second item
```
even if there are an arbitrary number of empty lines between the two items.

Reply:
ok, the 3 being nested into the 2 is definitely not what i would expect to happen when using the command, but to be honest i am not sure if i would expect to happen there if i were to use the commands (i just write markdown directly and don't use them) but probably to skip over the empty lines and to continue the numbering with the next non-empty line

Reply:
Okay should work now as expected :D